### PR TITLE
HBASE-23655 Fix flaky TestRSGroupsKillRS: should wait the SCP to finish

### PR DESCRIPTION
**Pull request by infraio**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 39s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 24s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 27s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   0m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m 47s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 11s |  hbase-rsgroup: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 32s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 54s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 50s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  52m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/996 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2cf6d35a0d99 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-996/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 6d22b7c255 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/1/artifact/out/diff-checkstyle-hbase-rsgroup.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/1/testReport/ |
| Max. process+thread count | 3961 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 54s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 42s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 24s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   0m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m 45s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 12s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 22s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 27s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 55s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  55m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/996 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 710a1fac4ca7 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-996/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 6d22b7c255 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/2/testReport/ |
| Max. process+thread count | 3977 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsKillRS.java [hunk 1]
```diff
@@ -237,12 +238,12 @@ public void testLowerMetaGroupVersion() throws Exception{
...
-    tableName = TableName.META_TABLE_NAME;
     Set<TableName> toAddTables = new HashSet<>();
-    toAddTables.add(tableName);
+    toAddTables.add(TableName.META_TABLE_NAME);
     rsGroupAdmin.moveTables(toAddTables, groupName);
-    assertTrue(rsGroupAdmin.getRSGroupInfo(groupName).getTables().contains(tableName));
-    TEST_UTIL.waitTableAvailable(tableName, 30000);
+    assertTrue(
+        rsGroupAdmin.getRSGroupInfo(groupName).getTables().contains(TableName.META_TABLE_NAME));
+    TEST_UTIL.waitUntilAllRegionsAssigned(TableName.META_TABLE_NAME, 30000);
```
Comment (Apache9) [inline hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsKillRS.java]: Does this method work for meta table? I'm not sure, just asking. As lots of the methods in HBTU just read the meta table so can not be used to test meta table itself...

File: hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsKillRS.java [hunk 2]
```diff
@@ -268,7 +269,9 @@ public void testLowerMetaGroupVersion() throws Exception{
     assertTrue(VersionInfo.compareVersion(originVersion,
         master.getRegionServerVersion(getServerName(servers.iterator().next()))) > 0);
     LOG.debug("wait for META assigned...");
-    TEST_UTIL.waitTableAvailable(tableName, 30000);
+    TEST_UTIL.waitFor(60000, () -> !TEST_UTIL.getHBaseCluster().getMaster().getProcedures().stream()
```
Comment (Apache9) [inline hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsKillRS.java]: So this is the root cause.

Comment (infraio) [reply hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsKillRS.java]: Yes. This is the root cause. In my test, the waitUntilAllRegionsAssigned and waitTableAvailable not worked for meta table...

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 36s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 25s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   0m 47s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m 46s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  9s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 24s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 24s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 23s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 56s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  52m 17s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/996 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 6d95c8ec68c8 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-996/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 6d22b7c255 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/3/testReport/ |
| Max. process+thread count | 3867 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-996/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**